fdbclient/ManagementAPI.actor.cpp

```diff
@@ -104,8 +104,8 @@ std::map<std::string, std::string> configForToken( std::string const& mode ) {
 	// Add any new store types to fdbserver/workloads/ConfigureDatabase, too
 
 	if (storeType.present()) {
-		out[p+"log_engine"] = format("%d", logType.get());
-		out[p+"storage_engine"] = format("%d", storeType.get());
+		out[p+"log_engine"] = format("%d", logType.get().operator KeyValueStoreType::StoreType());
+		out[p+"storage_engine"] = format("%d", storeType.get().operator KeyValueStoreType::StoreType());
```

Comment:
I don't like c-style casts because I have a hard time reading them. For c-style cast the compiler will try the following casts in order:

1. const_cast<new_type>(expression);
1. static_cast<new_type>(expression), with extensions: pointer or reference to a derived class is additionally allowed to be cast to pointer or reference to unambiguous base class (and vice versa) even if the base class is inaccessible (that is, this cast ignores the private inheritance specifier). Same applies to casting pointer to member to pointer to member of unambiguous non-virtual base;
1. static_cast (with extensions) followed by const_cast;
1. reinterpret_cast<new_type>(expression);
1. reinterpret_cast followed by const_cast.

I never am able to remember the actual ordering and I consider c-style casts as generally dangerous.

Reply:
However, I think a functional cast will work here...

Reply:
If the problem with the original is that calling the operator is ugly, could we define another function that isn't ugly to return the `StoreType`? That seems more clear than any option which is casting a `KeyValueStoreType` to a `StoreType`.

Reply:
Yes, I think this is the cleanest option. I will make this change as soon as I have some free cycles to work on this PR again.